plugins/tileacq.py

```diff
@@ -171,22 +176,32 @@ def _guess_smallest_fov(self):
           Note: they are not necessarily from the same FoV.
         raise ValueError: If no stream selected
         """
-        fovs = [self._get_fov(s) for s in self._get_streams()]
+        fovs = [self._get_fov(s) for s in self._get_live_streams(
+            self.main_app.main_data.tab.value.tab_data_model)]
         if not fovs:
             raise ValueError("No stream so no FoV, so no minimum one")
 
         return (min(f[0] for f in fovs),
                 min(f[1] for f in fovs))
 
     def show_dlg(self):
-        # TODO: only accept to run if in "secom_live" or "sparc_acqui" tab
-        # (iow, has a streambar_controller)
-        # TODO: support SPARC by using "sparc_acqui"
         # TODO: if there is a chamber, only allow if there is vacuum
 
-        tab = self.main_app.main_data.getTabByName("secom_live")
-        ss = tab.tab_data_model.streams.value
+        # Fail if the live tab is not selected
+        tab = self.main_app.main_data.tab.value
+        if tab.name not in ("secom_live", "sparc_acqui"):
+            box = wx.MessageDialog(self.main_app.main_frame,
+                       "Tiling must be done from the acquisition stream.",
+                       "Tiled acquisition not possible", wx.OK | wx.ICON_STOP)
+            box.ShowModal()
+            box.Destroy()
+            return
+
+        # During tiling, the user is interested in acquiring a large area of the
+        # sample, so the maximum roi is selected.
+        tab.tab_data_model.semStream.roi.value = (0, 0, 1, 1)
```

Comment:
When a roi that is smaller than the entire area is selected, I get an error. Since the user is probably not going to use tiling with small rois (then what's the point?), I just assumed that I can select the entire area whenever the tiling dialog is opened. Do you think we can do this or should I try to get it to work with custom roi sizes?

Reply:
Anyway, it should be 
if hasattr(tab.tab_data_model, "semStream"): ... 
to make it work on SECOM also